packages/replay/src/integration.ts

```diff
@@ -190,25 +201,45 @@ Sentry.init({ replaysOnErrorSampleRate: ${errorSampleRate} })`,
   /** Setup the integration. */
   private _setup(): void {
     // Client is not available in constructor, so we need to wait until setupOnce
-    this._loadReplayOptionsFromClient();
+    const finalOptions = this._loadReplayOptionsFromClient(this._initialOptions);
 
     this._replay = new ReplayContainer({
-      options: this._options,
+      options: finalOptions,
       recordingOptions: this._recordingOptions,
     });
   }
 
   /** Parse Replay-related options from SDK options */
-  private _loadReplayOptionsFromClient(): void {
+  private _loadReplayOptionsFromClient(initialOptions: InitialReplayPluginOptions): ReplayPluginOptions {
```

Comment:
l: Actually we are not using `this` here anymore, so we can extract this out into a separate function?

Reply:
so `function loadReplayOptionsFromClient(...)`

Reply:
Yup, sounds good!